src/std/misc/string.ss

```diff
@@ -254,3 +256,50 @@
    ((? (and number? inexact?) v) "~f")
    ((? (or list? hash-table? vector? ##values? obj-pr?) v) "~r")
    (else "~a")))
+
+;; Like CL SUBSTITUTE but specialized for strings and chars
+(def (string-substitute-char-if
+      newchar predicate string
```

Comment:
the argument order is all wrong here. The natural order in Scheme is `string newchar predicate`.

When something is called `string-do-something` the expectation is that the first argument is the string to work on.

Can  we fix this?

Reply:
Yes, we can "fix" that, but then it becomes harder to refer to the CL SUBSTITUTE as the model for this function.

Reply:
Done.